seq2science/rules/configuration_generic.smk

```diff
@@ -246,32 +246,39 @@ if "assembly" in samples:
+                    providers[assembly] = {"genome": None, "annotation": None}
+
+                    # check if the annotation AND genome can be downloaded
+                    if providers[assembly]["annotation"] is None:
+                        annotion_provider = provider_with_file("annotation", assembly)
+                        if annotion_provider:
+                            providers[assembly]["genome"] = annotion_provider  # genome always exists if annotation does
+                            providers[assembly]["annotation"] = annotion_provider
+
+                        # if the annotation cannot be downloaded, but exists locally, leave it be.
+                        elif any(os.path.exists(f) for f in [f"{file}.annotation.gtf", f"{file}.annotation.gtf.gz"]) and \
+                            any(os.path.exists(f) for f in [f"{file}.annotation.bed", f"{file}.annotation.bed.gz"]):
+                            providers[assembly]["annotation"] = "local"
+                            # also check if the genome exists when using a local annotation
                             if os.path.exists(f"{file}.fa"):
                                 providers[assembly]["genome"] = "local"
-                            if any(os.path.exists(f) for f in [f"{file}.annotation.gtf", f"{file}.annotation.gtf.gz"]) and \
-                                any(os.path.exists(f) for f in [f"{file}.annotation.bed", f"{file}.annotation.bed.gz"]):
-                                providers[assembly]["annotation"] = "local"
-
-                            # check if the annotation can be downloaded
-                            if providers[assembly]["annotation"] is None:
-                                annotion_provider = provider_with_file("annotation", assembly)
-                                if annotion_provider:
-                                    providers[assembly]["genome"] = annotion_provider  # genome always exists if annotation does
-                                    providers[assembly]["annotation"] = annotion_provider
-
-                            # check if the genome can be downloaded
-                            if providers[assembly]["genome"] is None:
-                                genome_provider = provider_with_file("genome", assembly)
-                                providers[assembly]["genome"] = genome_provider
+
+                    # check if the genome can be downloaded
+                    if providers[assembly]["genome"] is None:
+                        genome_provider = provider_with_file("genome", assembly)
+                        if genome_provider:
+                            providers[assembly]["genome"] = genome_provider
+                        # if the genome cannot be downloaded, but exists locally, leave it be.
+                        elif os.path.exists(f"{file}.fa"):
+                            providers[assembly]["genome"] = "local"
 
                     pickle.dump(providers, open(providersfile, "wb"))
```

Comment:
I guess we want to not include the "local" ones in the providersfile either?

Reply:
you're right! this dict only gets accessed when downloading, so local assemblies can just be skipped.